wire-runtime/src/main/java/com/squareup/wire/WireOutput.java

```diff
@@ -50,11 +50,12 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import java.io.IOException;
+import okio.BufferedSink;
 
 /**
  * Utilities for encoding and writing protocol message fields.
  */
-public final class WireOutput {
```

Comment:
So having done this, I'm not sure how much I like it. I think I'm going to back off and go the other way, turn it into `ProtoReader` and `ProtoWriter` (ala `JsonReader` and `JsonWriter`) and completely hide the semantics of the underlying `BufferedSink`.